From 7bb5f44e6367ebc5bd8abeba025b566329ca797a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 13 Nov 2024 20:25:00 +0100
Subject: [PATCH] application: Call setlocale() earlier to fix early i18n

Ensure that `setlocale()` is called earlier on in application
initialisation, so that the string passed to `g_set_application_name()` is
translated. Otherwise `setlocale()` is only called in `gtk_init()`, which
happens later, in the chain-up from `startup()`.
---
 src/application.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/application.c b/src/application.c
index 357d69c..0fecdb6 100644
--- a/src/application.c
+++ b/src/application.c
@@ -24,6 +24,7 @@
 #include <glib/gi18n.h>
 #include <gio/gio.h>
 #include <gtk/gtk.h>
+#include <locale.h>
 
 #include "application.h"
 #include "event-manager.h"
@@ -128,6 +129,7 @@ constructed (GObject *object)
 	g_application_set_flags (G_APPLICATION (object), G_APPLICATION_HANDLES_COMMAND_LINE);
 
 	/* Localisation */
+	setlocale (LC_ALL, "");
 	bindtextdomain (GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR);
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
-- 
GitLab

